### ISSUE TITLE
0.13: [Bug]: `garden test -n` backwards-incompatible

### ISSUE THREAD

**Issue by stefreak** [garden-io/garden#4178]
### Current Behavior

Not possible to run single tests by just the test name, if declared in a module.

```
garden test -n e2e
```

leads to error
```
✖ Test action "e2e" was not found.
```

### Expected behavior

If two tests in modules with names e2e are declared, I can run them using `garden test -n e2e`, like in 0.12

```
 % garden test -n e2e
Running tests 🌡️

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
🌍  Running in namespace default in environment default
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  cloud-dashboard           → Authorizing...
Unable to authenticate against Cloud Dashboard with the current session token.
Command results for this command run will not be available in Cloud Dashboard. If this not a
Cloud Dashboard project you can ignore this warning. Otherwise, please try logging out with
`garden logout` and back in again with `garden login`.
✔ providers                 → Getting status... → Cached
   ℹ Run with --force-refresh to force a refresh of provider statuses.
✔ graph                     → Resolving 2 modules... → Done
✔ test2                     → Building version v-c7c91aa62b... → Done (took 0 sec)
✔ test1                     → Building version v-c7c91aa62b... → Done (took 0 sec)
✔ test2                     → [echo]:  → Success (took 0.2 sec)
✔ test1                     → [echo]:  → Success (took 0.2 sec)

Done! ✔️
```

### Reproducible example

garden.yml
```yaml
kind: Project
name: repro
environments:
 - name: default
providers:
 - name: exec

---

kind: Module
type: exec
name: test1
include: []
tests:
  - name: e2e
    command: [echo]

---
kind: Module
type: exec
name: test2
include: []
tests:
  - name: e2e
    command: [echo]
```

### Workaround

prefix the test name with `*-`

```
garden test -n e2e -n '*-e2e'
```

### Suggested solution(s)

Automatically prefix strings passed to the deprecated `-n` parameter with `*-`

### Additional context

Found during dogfooding 0.13

### Your environment

* **OS:** macos
* **How I'm running Kubernetes:** EKS

`garden version`
0.13

**Comment by edvald** [comment]
@stefreak this isn't exactly a bug, but it's debatable what we want to allow here. Tests from modules are names as `<module name>-<test name>`, which is why your suggestion works (`*-e2e`). 

Maybe it makes sense to provide guidance in the command if no tests are found but the project contains modules? I think I'd prefer that over adding implicit behavior.

**Comment by edvald** [comment]
Also, the `--name/-n` parameter is deprecated, and does the exact same as positional args.

**Comment by stefreak** [comment]
Deprecated means for me it can be still used as before, but it shouldn't be used anymore – and not, it doesn't work like before anymore

**Comment by stefreak** [comment]
@edvald Basically any commands that might be used in workflows shouldn't have backwards incompatible changes, otherwise workflows are not backwards compatible. Are there more changes like this to commands?

**Comment by edvald** [comment]
Mmm I get your point, makes sense. We can change that back, specifically in this command and just the `--name` flag behavior specifically. Are you able to go for it? Or at least put it on the 0.13 project board.

**Comment by vvagaytsev** [comment]
Fixed in #4208
